go/engine/device_history.go

```diff
@@ -132,6 +133,23 @@ func (e *DeviceHistory) loadDevices() error {
 		e.devices = append(e.devices, exp)
 	}
 
+	// Load the last used times, but only if these are your own devices. The
+	// API won't give you those times for other people's devices.
+	if e.user.GetNormalizedName().Eq(e.G().Env.GetUsername()) {
+		lastUsedTimes, err := e.getLastUsedTimes()
+		if err != nil {
+			return err
+		}
+		for i := range e.devices {
+			detail := &e.devices[i]
+			lastUsedTime, ok := lastUsedTimes[detail.Device.DeviceID]
+			if !ok {
+				return fmt.Errorf("Failed to load last used time for device %s", detail.Device.DeviceID)
```

Comment:
So on the one hand aggressively returning an error here was what caused the test failure I just had to fix (revoked devices don't get a last_used_time from keys/fetch_private). On the other hand, I'm worried it would be *really* easy to break the code that loads this data without noticing.